src/wp-includes/option.php

```diff
@@ -253,6 +253,118 @@ function get_option( $option, $default_value = false ) {
 	return apply_filters( "option_{$option}", maybe_unserialize( $value ), $option );
 }
 
+/**
+ * Primes specific options into the cache with a single database query.
+ *
+ * Only options that do not already exist in cache will be primed.
+ *
+ * @since 6.4.0
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param array $options An array of option names to be primed.
+ */
+function prime_options( $options ) {
+	$alloptions     = wp_load_alloptions();
+	$cached_options = wp_cache_get_multiple( $options, 'options' );
+
+	// Filter options that are not in the cache.
+	$options_to_prime = array();
+	foreach ( $options as $option ) {
+		if ( ( ! isset( $cached_options[ $option ] ) || ! $cached_options[ $option ] ) && ! isset( $alloptions[ $option ] ) ) {
```

Comment:
Why do we need to check for both the existence of the option in the cached options group and also whether the value is falsey? I would think that we can skip priming any options that are found in either cached options or alloptions, eg:

```suggestion
		if ( ! ( isset( $cached_options[ $option ] ) || isset( $alloptions[ $option ] ) ) ) {
```

Reply:
@joemcgill We do have to check that the values are false-y, because for the return value of `wp_cache_get_multiple()`, cache misses will result in `false`, not `null`. However, that leads to a point that the `! isset( $cached_options[ $option ] )` check is technically unnecessary - although one might argue it's still useful from a perspective of defensive coding, since omitting it may _possibly_ result in a PHP warning.

The logic here adds options to the list of options to prime if their cache lookup was not successful _and_ they're not found in `$alloptions` either, so I think that is correct.